### PR TITLE
hmc5883 temperature compensation

### PR DESCRIPTION
**Pull request by tridge**
this also fixes the -C option to hmc5883 start for automatic scaling calibration on startup

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Kicked CI systems again.

**Comment by LorenzMeier** [comment]
So I assume that the way to roll this out would be to launch the temp compensation only on FMUv3 units, where a HMC5983 could be assumed. We can try to do it on FMUv2 in the driver instead of within the chip, but only if we learn the sensor-specific function for the board on a boot (because that's the only way to tell if someone connected a new or an old GPS, and we would risk to over-compensate if they would upgrade their GPS).

So in short: We can benefit on FMUv3 very easily by just enabling it on-chip, but fixing all use cases requires some pretty involved calibration and detection routines.

**Comment by tridge** [comment]
I've left it as a command line option so that PX4 can choose whether to enable it or not per board type. For ArduPilot I've decided to enable it unconditionally as it seems to do no harm on all the parts labeled 5883 that I've tested. The code also auto-disables it if it fails to read the temperature.
The harder decision will be once we have code to do temperature compensation in software for sensors. Should we tell people to get a real 5983, or should we build a table? 
If we do build a table then we should find the table has very small corrections for a real 5983 and large corrections for a 5883.

**Comment by LorenzMeier** [comment]
Thanks, applied!
